internal/azure/environments.go

```diff
@@ -0,0 +1,321 @@
+package internal
```

Comment:
I was not able to find a migration to a different library for these types which live in `go-autorest/autorest/azure`, but since there are no dependencies on external libraries/API calls I think it might be okay to have an internal package for these types.

We are only using the `Environment` struct, the `PublicCloud` variable, and the `EnvironmentFromName`  function so this can be thinned down further. I am mainly concerned about the licensing for this code, which states:

```
// Copyright 2017 Microsoft Corporation
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
```

Is this code okay to port internally?